### PR TITLE
dlb: update README

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
Add info on how to configure dlb driver and vfs.

-For commands that need superuser access, I used $ sudo than #, since # is used for comment and it does not look nice.
-Though the sysfs such as /sys/bus/pci/devices/0000:6d:00.0 may be different in another machine, I thought it would be more clear to just give a specific example than explaning.

Signed-off-by: Hyeongju Johannes Lee <hyeongju.lee@intel.com>

### CODE REVIEW

File: cmd/dlb_plugin/README.md [hunk 1]
```diff
@@ -19,6 +19,85 @@ Table of Contents
...
+$ sudo insmod dlb2.ko
+```
+
+Then, available dlb device nodes are visible in devfs.
+```bash
+$ ls -1 /dev/dlb*
+/dev/dlb0  /dev/dlb1  /dev/dlb2 ...
+```
+### VF configuration
+If you configure SRIO-V/VF (virtual functions), continue the following configurations.
```
Comment (mythi) [inline cmd/dlb_plugin/README.md:36]: SR-IOV

**Comment by bart0sh** [comment]
@hj-johannes-lee should we also describe PF(physical function) setup?

File: cmd/dlb_plugin/README.md [hunk 2]
```diff
@@ -19,6 +19,85 @@ Table of Contents
...
+$ sudo echo 2048 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_atomic_inflights
+$ sudo echo 2048 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_dir_credits
+$ sudo echo 64 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_dir_ports
+$ sudo echo 2048 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_hist_list_entries
+$ sudo echo 8192 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_ldb_credits
+$ sudo echo 64 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_ldb_ports
+$ sudo echo 32 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_ldb_queues
+$ sudo echo 32 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_sched_domains
+$ sudo echo 2 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_sn0_slots
+$ sudo echo 2 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_sn1_slots
```
Comment (bart0sh) [inline cmd/dlb_plugin/README.md:94]:
```suggestion
$ echo 2048 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_atomic_inflights
$ echo 2048 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_dir_credits
$ echo 64 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_dir_ports
$ echo 2048 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_hist_list_entries
$ echo 8192 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_ldb_credits
$ echo 64 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_ldb_ports
$ echo 32 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_ldb_queues
$ echo 32 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_sched_domains
$ echo 2 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_sn0_slots
$ echo 2 | sudo tee -a /sys/bus/pci/devices/0000\:6d\:00.0/vf0_resources/num_sn1_slots
```

File: cmd/dlb_plugin/README.md (hunk 1 above)
Comment (hj-johannes-lee) [reply cmd/dlb_plugin/README.md:36]: Oops, it's from the wiki. let me change both.!

**Comment by bart0sh** [comment]
@hj-johannes-lee can we add DLB info to the https://github.com/intel/intel-device-plugins-for-kubernetes/tree/main/demo as well? It would also make sense to mention demo images & pod yamls here as a alternative (and faster) way of running DLB tests.

**Comment by hj-johannes-lee** [comment]
> @hj-johannes-lee can we add DLB info to the https://github.com/intel/intel-device-plugins-for-kubernetes/tree/main/demo as well? It would also make sense to mention demo images & pod yamls here as a alternative (and faster) way of running DLB tests.

Do I need to make demo videos also?

**Comment by bart0sh** [comment]
What I meant is that our users should know that there is a less manual way of testing DLB. You're explaining here fully manual way, but some of this steps are already automated in our demo Dockerfiles. For example, instead of downloading libdlb sources and building them manually users may want to simply apply [this yaml](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/demo/dlb-libdlb-demo-pod.yaml) or run [this image](https://github.com/intel/intel-device-plugins-for-kubernetes/tree/main/demo/dlb-libdlb-demo)

**Comment by bart0sh** [comment]
> Do I make a video?

Up to you. This is probably out of scope of this PR though.

**Comment by hj-johannes-lee** [comment]
Sorry, I think.. I am now just too obsessed with the expression "faster," so I do not understand exactly..

But,, since configuration is necessary for a user even if the user will use the docker image or yaml file for test,, 

the actual difference would be..
cd ../../libdlb/ && make && sudo LD_LIBRARY_PATH=$PWD ./examples/dir_traffic -n 128
vs
kubectl apply -f something for dlb device plugin
kubectl apply -f something for dlb-libdlb-demo.yaml

Just I do not know why it is faster..!

And.. Isn't it enough to have the [testing-the-plugin part](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/dlb_plugin/README.md#testing-the-plugin)..?


> It would also make sense to mention demo images & pod yamls here as a alternative (and faster) way of running DLB tests.

As I asked already, I do not understand whether you meant the readme file in dlb device plugin, or readme file in demo...

**Comment by bart0sh** [comment]
> Sorry, I think.. I am now just too obsessed with the expression "faster," so I do not understand exactly..

Don't bee. You can mention this as an alternative way if you don't think it's faster.

File: cmd/dlb_plugin/README.md [hunk 3]
```diff
@@ -19,6 +19,137 @@ Table of Contents
...
+```bash
+$ sudo ./usertools/dpdk-devbind.py --status | grep -A3 'Eventdev devices using DPDK-compatible driver'
+```
+
+Run eventdev test for the PF device:
+```bash
+$ sudo ./builddir/app/dpdk-test-eventdev --no-huge -a '6d:00.0' -- --test=order_queue --plcores=1 --wlcore=2,3 --nb_flows=64
+```
+
+> **Note:**: Alternative and faster way is to use Dockerfile from this [PR](https://github.com/intel/intel-device-plugins-for-kubernetes/pull/733) for patching and building DPDK and running tests.
```
Comment (bart0sh) [inline cmd/dlb_plugin/README.md:80]:
```suggestion
> **Note:**: Alternative and faster way is to use [Dockerfile](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/demo/dlb-dpdk-demo/Dockerfile) for patching and building DPDK and running tests.

Comment (mythi) [reply cmd/dlb_plugin/README.md:80]: This is enough for now and we don't need to add anything to `demo/readme.md`.

File: cmd/dlb_plugin/README.md [hunk 4]
```diff
@@ -19,6 +19,137 @@ Table of Contents
...
+
+Make sure that DLB device is in the list of eventdev devices known to dpdk and handled by vfio-pci driver:
+```bash
+$ sudo ./usertools/dpdk-devbind.py --status | grep -A3 'Eventdev devices using DPDK-compatible driver'
+```
+
+Run eventdev test for the PF device:
+```bash
+$ sudo ./builddir/app/dpdk-test-eventdev --no-huge -a '6d:00.0' -- --test=order_queue --plcores=1 --wlcore=2,3 --nb_flows=64
+```
```
Comment (mythi) [inline cmd/dlb_plugin/README.md:78]: DLB plugin does not support `vfio-pci` devices. Why do we have it here?

**Comment by hj-johannes-lee** [comment]
I changed some structure because I thought building dpdk is only for running demo and it makes sense to separate from the part that explains about the "configuration" of pf/vf "devices."

I hope (and believe) that it still satisfies all of your suggestions and it looks more logical than before.!

**Review by mythi — APPROVED** [review]
LGTM

**Review by bart0sh — APPROVED** [review]
/lgtm

**Pull request merged**